alphadia/workflow/optimization.py

```diff
@@ -260,6 +234,54 @@ def _check_convergence(self):
             < 1.1 * self.history_df[self.feature_name].iloc[-3]
         )
 
+    def _find_index_of_optimum(self):
+        """Finds the index of the row in the history dataframe with the optimal value of the feature used for optimization.
+
+        Returns
+        -------
+        int
+            The index of the row with the optimal value of the feature used for optimization.
+
+        Notes
+        -----
+            This method may be overwritten in child classes if the "optimal" value differs from the value that maximizes the feature used for optimization.
+
+        """
+        return self.history_df[self.feature_name].idxmax()
+
+    def _update_optimization_manager(self):
+        """Updates the optimization manager with the results of the optimization, including:
+            the optimal parameter
+            the classifier version,
+            score cutoff, and
+            FWHM values
+        at the optimal parameter.
+
+        """
+        index_of_optimum = self._find_index_of_optimum()
+
+        optimal_parameter = self.history_df["parameter"].loc[index_of_optimum]
+        classifier_version_at_optimum = self.history_df["classifier_version"].loc[
+            index_of_optimum
+        ]
+        score_cutoff_at_optimum = self.history_df["score_cutoff"].loc[index_of_optimum]
+        fwhm_rt_at_optimum = self.history_df["fwhm_rt"].loc[index_of_optimum]
+        fwhm_mobility_at_optimum = self.history_df["fwhm_mobility"].loc[
+            index_of_optimum
+        ]
+
+        self.workflow.optimization_manager.fit({self.parameter_name: optimal_parameter})
+        self.workflow.optimization_manager.fit(
+            {"classifier_version": classifier_version_at_optimum}
+        )
+        self.workflow.optimization_manager.fit(
+            {"score_cutoff": score_cutoff_at_optimum}
+        )
+        self.workflow.optimization_manager.fit({"fwhm_rt": fwhm_rt_at_optimum})
+        self.workflow.optimization_manager.fit(
+            {"fwhm_mobility": fwhm_mobility_at_optimum}
+        )
```

Comment:
alternative ordering:

```python
optimal_parameter = self.history_df["parameter"].loc[index_of_optimum]
self.workflow.optimization_manager.fit({self.parameter_name: optimal_parameter})

classifier_version_at_optimum = ...
self.workflow.optimization_manager.fit(
    {"classifier_version": classifier_version_at_optimum}
)
...

```
unless, of course, the `.fit()` operations interfere with the `*_at_optimum` values